### DISCUSSION TITLE
How do I enable cgroups-v2

### DISCUSSION THREAD

**Discussion by lunacd** [nix-community/NixOS-WSL#508] [General]
I'm getting the idea that cgroups-v2 should be the default now but I'm not sure why `podman` is still complaining. The suggestions online mostly point to adding kernel parameter to disable cgroups-v1: `kernelCommandLine = cgroup_no_v1=all`.

Should I need to do that to make podman use cgroups-v2 in the first place? If so, how would I add an extra configuration option to `wsl.conf` that is not builtin?

**Comment by SuperSandro2000** [comment]
If you are using WSL2 with systemd and 24.05 then it should be automatically enabled.

**Reply by lunacd** [reply]
I'm on systemd, but I initially installed 23.11 and upgraded to 24.05, i.e. `stateVersion = "23.11"`. Would that affect things? If it does, how do I get the new default behavior?

**Reply by SuperSandro2000** [reply]
No, that doesn't matter

**Reply by 573** [reply]
Check that you have a recent enough wsl version, do a `wsl --upgrade` if in doubt.

**Reply by lunacd** [reply]
Thanks! But I have the latest version installed:
```
Checking for updates.
The most recent version of Windows Subsystem for Linux is already installed.
```

Maybe it's my podman that's configured incorrectly here?

**Reply by lunacd** [reply]
Oh actually, when I run `grep cgroup /proc/filesystems`, I do see `cgroup2` there. So I should have been looking at what's wrong with my podman's configuration from the start
